T3/Gui/UiHelpers/UIControlledSetting.cs

```diff
@@ -0,0 +1,92 @@
+using ImGuiNET;
+using System;
+using T3.Gui.UiHelpers;
+
+namespace T3.Gui.Windows
+{
+    public class UIControlledSetting
+    {
+        /// <summary>
+        /// The generated unique label
+        /// </summary>
+        private string _uniqueLabel;
+        private string _hiddenUniqueLabel;
+
+        /// <summary>
+        /// The label provided in the constructor
+        /// </summary>
+        public string CleanLabel { get; private set; }
+        public bool DrawOnLeft { get; private set; }
+
+        private string _tooltip;
+        private string _additionalNotes;
+        private Func<string, bool> _guiFunc;
+        private Action _OnValueChanged;
+
+        // Cheaper than GUIDs
+        // In case we want to have the same variable be changeable with different UI controls
+        // or if multiple settings have the same label
+        static ushort _countForUniqueID = ushort.MaxValue;
+
+        /// <summary>
+        /// For the sake of simple use of the optional parameters and populating/maintaining many settings, the recommended way to call this constructor is:
+        /// <code>
+        /// new UIControlledSetting
+        /// (
+        ///    label: "My Setting",
+        ///    tooltip: "The global scale of all rendered UI in the application",
+        ///    guiFunc: (string guiLabel) => CustomComponents.FloatValueEdit(guiLabel, ref UserSettings.Config.UiScaleFactor, 0.01f, 0.5f, 3f),
+        ///    OnValueChanged: () => //your action
+        /// );
+        /// </code>
+        /// </summary>
+        /// <param name="label">The label to display next to the gui control</param>
+        /// <param name="guiFunc">The <see cref="ImGuiNET"/> - based function that draws the setting control and
+        /// returns true if the control was changed. The input to this function see is a unique ID based on the label provided</param>
+        /// <param name="tooltip">Tooltip displayed when hovering over the control</param>
+        /// <param name="additionalNotes">Additional notes displayed alongside the tooltip [Not currently in use, but will be once T3 tooltips are integrated]</param>
+        /// <param name="OnValueChanged">An action performed when the value is changed</param>
+        public UIControlledSetting(string label, Func<string, bool> guiFunc, string tooltip = null, string additionalNotes = null, bool drawOnLeft = false, Action OnValueChanged = null)
+        {
+            CleanLabel = label;
+            _hiddenUniqueLabel = $"##{label}{_countForUniqueID--}";
+            _uniqueLabel = $"{label}##{_countForUniqueID--}";
+
+            _guiFunc = guiFunc;
+            _tooltip = tooltip;
+            _additionalNotes = additionalNotes;
+            _OnValueChanged = OnValueChanged;
+            DrawOnLeft = drawOnLeft;
+        }
+
```

Comment:
im interested in your take on this, as to me this seems like a good way to enable Tooll contributors to add/remove user-facing settings with ease. i'm not deeply familiar with the model vs view concept though i think i know what you mean and share the sentiment in most cases. this case to me acts entirely as "view" - just a simple and consistent way to do so. Though with `OnValueChanged` it does allow that scope to creep

I like construction with object initializers too, but I chose to go this route due to the auto-generated unique labeling system (used to prevent certain [imgui bugs](https://github.com/ocornut/imgui/issues/74) that can be easy to run into). You'll notice that when making use of this constructor elsewhere, I essentially treat it like an object initializer, especially due to the optional parameters. Like this:

``
            new UIControlledSetting
            (
                label: "UI Scale",
                tooltip: "The global scale of all rendered UI in the application",
                guiFunc: (string guiLabel) => CustomComponents.DrawSingleValueEdit..... etc
            ),
```

I also want to make sure that anyone creating settings can't mess it up - they know they need at least two things (a label and a draw function), and if they have that it cant go wrong